src/Cake.Core/Polyfill/EnvironmentHelper.cs

```diff
@@ -119,7 +119,28 @@ public static Runtime GetRuntime()
         public static FrameworkName GetBuiltFramework()
         {
 #if NETCORE
-            return new FrameworkName(".NETStandard,Version=v2.0");
+            if (netCoreAppFramwork != null)
+            {
+                return netCoreAppFramwork;
+            }
+
+            var assemblyPath = typeof(System.Runtime.GCSettings)?.GetTypeInfo()?.Assembly?.CodeBase;
+            if (string.IsNullOrEmpty(assemblyPath))
+            {
+                return NetStandardFramework;
+            }
+
+            const string microsoftNetCoreApp = "Microsoft.NETCore.App";
+            var runtimeBasePathLength = assemblyPath.IndexOf(microsoftNetCoreApp) + microsoftNetCoreApp.Length + 1;
+            var netCoreAppVersion = string.Concat(assemblyPath.Skip(runtimeBasePathLength).Take(3));
+            if (string.IsNullOrEmpty(netCoreAppVersion))
+            {
+                return NetStandardFramework;
+            }
+
+            return netCoreAppFramwork = Version.TryParse(netCoreAppVersion, out var version)
+                                            ? new FrameworkName(".NETCoreApp", version)
+                                            : NetStandardFramework;
```

Comment:
Rebase gone bad, or why is this change here? Didn't @devlead write this in another commit?